File: stage-common/src/main/java/io/github/solusmods/eternalcore/stage/impl/StageRegistry.java
package io.github.solusmods.eternalcore.stage.impl;

import dev.architectury.registry.registries.Registrar;
import dev.architectury.registry.registries.RegistrarManager;
import io.github.solusmods.eternalcore.stage.EternalCoreStage;
import io.github.solusmods.eternalcore.stage.ModuleConstants;
import io.github.solusmods.eternalcore.stage.api.Stage;
import lombok.AccessLevel;
import lombok.NoArgsConstructor;
import net.minecraft.core.Registry;
import net.minecraft.resources.ResourceKey;
import net.minecraft.resources.ResourceLocation;

@NoArgsConstructor(access = AccessLevel.PRIVATE)
@SuppressWarnings("unchecked")
public class StageRegistry {

    private static final ResourceLocation registryId = EternalCoreStage.create("stages");
    // endregion
    public static final Registrar<Stage> STAGES = RegistrarManager.get(ModuleConstants.MOD_ID).<Stage>builder(registryId)
            .syncToClients().build();
    public static final ResourceKey<Registry<Stage>> KEY = (ResourceKey<Registry<Stage>>) STAGES.key();




    public static void init() {}
}

File: stage-common/src/main/kotlin/io/github/solusmods/eternalcore/stage/impl/StageRegistry.kt
package io.github.solusmods.eternalcore.stage.impl

import dev.architectury.registry.registries.Registrar
import dev.architectury.registry.registries.RegistrarManager
import io.github.solusmods.eternalcore.stage.EternalCoreStage
import io.github.solusmods.eternalcore.stage.ModuleConstants
import io.github.solusmods.eternalcore.stage.api.Stage
import lombok.AccessLevel
import lombok.NoArgsConstructor
import net.minecraft.core.Registry
import net.minecraft.resources.ResourceKey
import net.minecraft.resources.ResourceLocation

@NoArgsConstructor(access = AccessLevel.PRIVATE)
object StageRegistry {
    private val registryId: ResourceLocation = EternalCoreStage.create("stages")

    // endregion
    val STAGES: Registrar<Stage?> = RegistrarManager.get(ModuleConstants.MOD_ID).builder<Stage?>(registryId)
        .syncToClients().build()
    val KEY: ResourceKey<Registry<Stage?>?>? = STAGES.key() as ResourceKey<Registry<Stage?>?>?


    fun init() {}
}
